### ISSUE TITLE
Android application not recording tree addresses for mapped trees

### ISSUE THREAD

**Issue by hadleystein1** [OpenTreeMap/otm-android#285]
Customer (Leaf Limb Consulting) has been mapping thousands of trees in the field. All trees on their map (https://www.opentreemap.org/durhamtestmap/map/) have been individually over the past few weeks using our Android application. The addresses for certain trees are not populating. Over 150 trees on their map do not contain addresses. Customer did not detect any pattern to when address were populating versus when addresses were not populating. I have attached the most recent export from their map.

In addition to fixing the issue, customer wants to know is there a way for addresses to retroactively populate so all tree records on the map contain address information? 

[LeafLimbConsulting_10_28.xlsx](https://github.com/OpenTreeMap/otm-android/files/558441/LeafLimbConsulting_10_28.xlsx)

**Comment by RickMohr** [comment]
OTM currently does not support retroactive geocoding. Currently we'd have to do it for them as a geocoding project.

A cheap OTM enhancement would be for "Quick Edit" to look up the address if it's missing. They'd have to select each of the 150 trees but that would go pretty fast.

**Comment by maurizi** [comment]
The Android app does not use reverse geocoding to populate the address field when adding a tree. We would have to add that functionality to the app (or possibly server-side).

**Comment by hadleystein1** [comment]
To clarify:
- The application is supposed to populate the address field? 
- As for finding the addresses of the already-mapped trees without address information that is not possible as the system is currently designed and/or without allocating additional time/resources to those specific records?

**Comment by maurizi** [comment]
> The application is supposed to populate the address field? 

The Android app does not currently do this, but the website does and according to @RickMohr the iOS app does as well.

> As for finding the addresses of the already-mapped trees without address information that is not possible as the system is currently designed and/or without allocating additional time/resources to those specific records? 

Correct

**Comment by hadleystein1** [comment]
Client said: _"We have been using Androids the entire time. Like I said, it only happened on two phones for one day during the three weeks."_ Is this possible? 

@jmorrison1847 does the Texas A&M have an address tool for getting addresses of Lat and Long information?

**Comment by jmorrison1847** [comment]
@hadleystein1 yes, up to 2,500 records: http://geoservices.tamu.edu/Services/Geocode/BatchProcess/

**Comment by maurizi** [comment]
I was wrong, the Android app does reverse geocode to fill in the address fields when adding a tree.  I don't know why some trees do not have the address information filled in.  I would assume it is because reverse geocoding failed to find an address (which usually happens if the point is not near any address).

**Comment by jmorrison1847** [comment]
^if that's the case, I'm not sure that there is a clear solution--Google's geocoding API is about as good as it gets.

**Comment by RickMohr** [comment]
We use ESRI's geocoder.

**Comment by maurizi** [comment]
@RickMohr Android uses the Google geocoder (because it is conveniently built-in)

**Comment by RickMohr** [comment]
Before assuming it's a problem with the geocoder I think we should try geocoding the lat/long for a few of the no-address trees.
